toolkit/tools/internal/grub/grubtokenizer.go

```diff
@@ -0,0 +1,507 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Parses a grub config file into tokens.
+// This handles proper interpretation of metacharacters, newlines, variable expansions, character escapes, single quoted
+// strings, and double quoted strings. While this is not a full parser into an AST (abstract syntax tree), it is better
+// than parsing the grub config files using regexes.
+
+package grub
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/grub/filescanner"
+)
+
+func TokenizeGrubConfigFile(name string) ([]Token, error) {
```

Comment:
nit: since this is already a part of `grub` package, we can consider dropping grub from the function names.

Reply:
Fixed.